From 94d5b95f18a687e2b5a38e856aa979467b9250e5 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 9 Sep 2013 17:02:36 +0300
Subject: [PORT FROM MAIN] Telephony: Fix call not resuming on sending mmi code "1"

BZ: 137025

If the user dials "1" When there is an active call and
a waiting call, then the active call is ended but the
waiting call is not accepted.

As per 3GPP 22.030, if the user dials "1" when there is
an active call and a waiting call, then the active call
needs to be released and the waiting call needs to be
accepted.

If there is a background or ringing call, then call
hangupForegroundResumeBackground or else call
hangupAllConnections.

Change-Id: If77247a5471b3d2700f2aa644eabfb25b85aedbd
Orig-Change-Id: I7bf6c54b84750b769111c0503ad107dd0211bc09
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
---
 .../internal/telephony/gsm/GsmCallTracker.java     |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
index 583b28c..6101c20 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmCallTracker.java
@@ -764,7 +764,7 @@ public final class GsmCallTracker extends CallTracker {
                     log("(foregnd) hangup dialing or alerting...");
                 }
                 hangup((GsmConnection)(call.getConnections().get(0)));
-            } else if (mBackgroundCall.isIdle()) {
+            } else if (mBackgroundCall.isIdle() && mRingingCall.isIdle()) {
                 hangupAllConnections(call);
             } else {
                 hangupForegroundResumeBackground();
-- 
1.7.4.1

